### DISCUSSION TITLE
create new account only by admin

### DISCUSSION THREAD

**Discussion by vamseekrishna25** [noobaa/noobaa-core#6450] [Q&A]
is there any option to allow only admin from nooba to create accounts

**Comment by nimrod-becker** [comment]
hi @vamseekrishna25 
Are you using the web UI for that?
In general for the web UI only accounts marked with access to the web UI can login and create new accounts.

Or are you referring to a different flow?

**Reply by vamseekrishna25** [reply]
Hi @nimrod-becker 
Not web UI I am asking creating new accounts through RPC  account_api create_account method it doesn't have any auth for creating an account.

**Reply by nimrod-becker** [reply]
Yes, that's a good and simple fix. We can verify that an account has admin rights (which are marked by the UI access flag) and only then allow account creation. S3 only accounts won't be able to do so.

@guymguym @romayalon am I missing some dependency here?

**Reply by romayalon** [reply]
@nimrod-becker  we need that operator account will be able to create accounts.

**Comment by nimrod-becker** [comment]
True, but we have 
```
 roles: {
                        type: 'array',
                        items: {
                            type: 'string',
                            enum: ['admin', 'user', 'viewer', 'operator']
                        }
                    },
```
So we can set account creation to be available for admin and operator

**Comment by nimrod-becker** [comment]
Fixed with https://github.com/noobaa/noobaa-core/pull/6520
